fastparse/shared/src/test/scala/fastparse/BmpTests.scala

```diff
@@ -99,16 +98,14 @@ object BmpTests extends TestSuite {
       )
     }
 
-    val bmp = P( fileHeader ~ header.flatMap {
-      case header: BitmapHeader =>
-        val infoPart = header.infoPart
-        bmpRow(infoPart.width, infoPart.bitsPerPixel)
-          .rep(exactly=infoPart.height)
-          .map(pixels => (header, pixels))
-    } ).map{
-      case (fileHeader, (bitmapHeader, pixels)) =>
-        Bmp(fileHeader, bitmapHeader, pixels.reverse)
-    }
+    val bmp = P(
+      for{
+        fileHeaderData <- fileHeader
+        headerData <- header
+        pixels <- bmpRow(headerData.infoPart.width, headerData.infoPart.bitsPerPixel)
```

Comment:
Looks great! But it's a little big confusing that the ordering of for inner statements represents ordering of subparsers in parser.